gui/ui/ui.c

```diff
@@ -727,17 +732,37 @@ void vui_textedit_input(vui_context_t *ctx, int index, const char *text)
     // Copy first part
     size_t old_len = (end - edit->text);
     char tmp[sizeof(edit->text)];
+    
+    // Ensure old_len doesn't exceed buffer size
```

Comment:
we already know sizeof(edit->text) can accomodate the entire concatenated string, so tmp is large enough